### PR TITLE
RavenDB-12145 PR Comments + Tests + fixes

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                        yObject = yLazyStringValue2;
+                        break;
+                    }
+                    return LogMissmatchTypesReturnOrder(xResult, yResult);
+                    break;
+                case LazyNumberValue xLazyNumberValue:
+                    xObject = xLazyNumberValue;
+                    switch (yResult)
+                    {
+                        case int i:
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]:
```
case int _:
case long _:
  yObject = yResult;
  break;
```
Is easier

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                    }
+                    return LogMissmatchTypesReturnOrder(xResult, yResult);
+                case long xLong:
+                    if (yResult is long yLong)
+                    {
+                        xObject = xLong;
+                        yObject = yLong;
+                        break;
+                    }
+                    return LogMissmatchTypesReturnOrder(xResult, yResult);
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]: And if it is double, we should still support it.

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                    }                                        
+                    break;
+                case int xInt:
+                    if (yResult is int yInt)
+                    {
+                        xObject = xInt;
+                        yObject = yInt;
+                        break;
+                    }
+                    return LogMissmatchTypesReturnOrder(xResult, yResult);
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]: and if it is double, long?

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+                            return xLazyNumberValue.CompareTo(lnv) * _order;
+                        default:
+                            return LogMissmatchTypesReturnOrder(xObject, yObject);
+                    }
+                case long xLong:
+                    if (yObject is long yLong)
+                    {
+                        return xLong.CompareTo(yLong) * _order;
+                    }
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]: double ?

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+                case IComparable xComparable:
+                    if(yObject.GetType() == xObject.GetType())
+                        return xComparable.CompareTo(yObject) * _order;
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]: long -> double, etc

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+                case BlittableJsonReaderArray xBlittableJsonReaderArray:
+                    if (yObject is BlittableJsonReaderArray yBlittableJsonReaderArray)
+                    {
+                        return string.CompareOrdinal(xBlittableJsonReaderArray.ToString(), yBlittableJsonReaderArray.ToString());
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]: You are allocating what may be very large value.
Need to recurse here.

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+                    }
+                    return LogMissmatchTypesReturnOrder(xObject, yObject);
+                case BlittableJsonReaderObject xBlittableJsonReaderObject:
+                    if (yObject is BlittableJsonReaderObject yBlittableJsonReaderObject)
+                    {
+                        return string.CompareOrdinal(xBlittableJsonReaderObject.ToString(), yBlittableJsonReaderObject.ToString());
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]: There is no meaningful way to compare them, and I certainly don't want to use strings.
Use the `Base` as a way to get just a stable sort order.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs (hunk 1 above)
Comment (talweiss1982) [reply src/Raven.Server/Documents/Queries/GraphQueryOrderByFieldComparer.cs]: its actually does the conversion

**Comment by talweiss1982** [comment]
updated

**Comment by ayende** [comment]
Also, check the failing tests

**Review by ayende — APPROVED** [review]

**Comment by ayende** [comment]
There is a failing test that needs to be handled.

**Comment by ayende** [comment]
There are also conflicts from my recent merge, please fix them and re-sumbit the PR

**Comment by talweiss1982** [comment]
I have resolved conflicts in my PR are there new modification you made?
 Which test is failing i have passed fast test and on slow test only SQL test failed from what i recall.

**Comment by arekpalinski** [comment]
Probably `SlowTests.Graph.SortGraphQueries.SortOnObjectShouldWork` test - http://jenkins.hibernatingrhinos.com/job/PR_Tests_v4.2/184/

**Comment by talweiss1982** [comment]
@arekpalinski 
I was afraid this test might be flaky.
Since now we don't really do a real comparison on blittable might as well delete the test since we can't know the order of objects.

**Comment by ayende** [comment]
Yes, we have a stable sort on the _existing_ system, but never between runs, so that might certainly be it.
Can you remove the test and reissue the PR?

**Comment by talweiss1982** [comment]
@ayende I'm just running the test and will submit a fixed PR once they pass

**Comment by talweiss1982** [comment]
acouple of tests are failing i might have done something wrong in the merge, checking.

**Pull request closed without merging**